src/core/producer/stage.h

```diff
@@ -82,6 +83,11 @@ class stage_base
     virtual std::future<frame_transform> get_current_transform(int index)                                          = 0;
     virtual std::future<void>
     load(int index, const spl::shared_ptr<frame_producer>& producer, bool preview = false, bool auto_play = false) = 0;
```

Comment:
It would be nicer to replace these other methods, instead of adding more with extra parameters.
The new parameter could have a default value of none, so to the couple of callsites that havent been touched in this PR could continue that way